pandas/core/indexes/api.py

```diff
@@ -213,14 +215,41 @@ def conv(i):
 
     if kind == "special":
         result = indexes[0]
+        first = result
```

Comment:
can you move all of this stuff out of api.py (e.g. new module), can be followon

Reply:
will give some thought to a proper home